### PR TITLE
[IMP] essentials: search filter and group

### PR DESCRIPTION
**Pull request by samueljlieber**
Task: [#3843408](https://www.odoo.com/web#id=3843408&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

Improves the Search and Filter documentation for 17.0 with the following improvements:

- Renamed h1 to Search, filter, and group records
- Removed any pronoun use
- Updated content and screenshots for 17.0
- Expanded the Custom Filters section with details
- Added the Favorites section

### CODE REVIEW

**Comment by samueljlieber** [comment]
Hi @tiku-odoo, this PR is ready for a peer review whenever you have a chance! Thank you :)

**Review by tiku-odoo — APPROVED** [review]
Hi @samueljlieber 

Great work on this Search/Filter PR! It's a huge improvement. 

I've made a handful of suggestions, some grammar/spelling, and then some clarifications. 

What do you think about adding some highlights to the images? If not, they work as is, just thought they might direct the "quick glance users" to the feature's location better. 

Great, work. Let me know if you have any questions. I've approved this PR. 

Thanks, Tim :+1:

File: content/applications/essentials/search.rst [hunk 1]
```diff
@@ -1,100 +1,206 @@
...
-clicking :guilabel:`Apply`.
+If the :ref:`preconfigured filters <search/preconfigured-filters>` are not specific enough, add a
+custom filter. To do so, click the :guilabel:`🔽 (down arrow)` icon in the search bar, then select
+:menuselection:`Filters --> Add Custom Filter`.
 
-.. example::
-   You could *only* include records from a single salesperson on the Sales Analysis report, for
-   example *Mitchell Admin*, by selecting :guilabel:`Salesperson` as the field, :guilabel:`is equal
-   to` as the operator, and typing `Mitchell Admin` as the value.
+The :guilabel:`Add Custom Filter` modal displays the matching option, filter rule, and a toggle to
```
[the remark this answers is not in the document]
Comment (samueljlieber) [reply content/applications/essentials/search.rst:84]: The term is correct here, _modal_ is the UI term for the pop-up window on the screen. However, we are still deciding on whether or not to use this terminology. So I will be changing this to "pop-up window" for now :)

**Comment by samueljlieber** [comment]
Thank you for your helpful review @tiku-odoo, I implemented your suggestions and added an example for branch merges. Passing to @ksc-odoo for final content review 👍

File: content/applications/essentials/search.rst [hunk 2]
```diff
@@ -1,100 +1,214 @@
...
+=================================
+Search, filter, and group records
+=================================
 
-Odoo uses filters to include only the most relevant records depending on the purpose of the view you
-are on. However, you can edit the default filter or search for specific values.
+Odoo allows for the searching, filtering, and grouping of records in a view to display only the most
+relevant records. The search bar is located at the top of the view, start typing to :ref:`search for
+values <search/values>`, or click the :guilabel:`🔽 (down arrow) icon` to access the :ref:`Filter
+<search/filters>`, :ref:`Group By <search/group>` and :ref:`Favorites <search/favorites>` drop-down
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:8]:
```suggestion
<search/filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites <search/favorites>` drop-down
```

File: content/applications/essentials/search.rst [hunk 3]
```diff
@@ -1,100 +1,214 @@
...
+menus.
+
+.. _search/values:
+
+Search for values
+=================
+
+Use the search field to quickly look for specific values and add them as a filter. Type the value to
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:16]:
```suggestion
Use the search field to quickly look for specific values, and add them as a filter. Type the value to
```

File: content/applications/essentials/search.rst [hunk 4]
```diff
@@ -1,100 +1,214 @@
...
+search for and select the desired option from the drop-down menu to apply the search filter.
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:17]:
```suggestion
search for, and select the desired option from the drop-down menu to apply the search filter.
```

File: content/applications/essentials/search.rst [hunk 5]
```diff
@@ -1,100 +1,214 @@
...
+
+.. example::
+   Instead of adding a :ref:`custom filter <search/custom-filters>` to select records where
+   *Mitchell Admin* is the salesperson on the *Sales Analysis* report (:menuselection:`Sales -->
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:21]:
```suggestion
   *Mitchell Admin* is the salesperson on the *Sales Analysis* report (:menuselection:`Sales app -->
```

File: content/applications/essentials/search.rst [hunk 6]
```diff
@@ -1,100 +1,214 @@
...
+   Reporting --> Sales`), search for `Mitch` and click the :guilabel:`⏵ (right arrow)` next to
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:22]:
```suggestion
   Reporting --> Sales`), search for `Mitch`, and click the :guilabel:`⏵ (right arrow)` next to
```

File: content/applications/essentials/search.rst [hunk 7]
```diff
@@ -1,100 +1,214 @@
...
+   :guilabel:`Search Salesperson for: Mitch`, and select :guilabel:`Mitchell Admin`.
+
+   .. image:: search/search-values.png
+      :align: center
+      :alt: Searching for a specific value on the Sales Analysis report
+
+.. note::
+   Using the search field is equivalent to using the *contains* operator when adding a :ref:`custom
+   filter <search/custom-filters>`. If a partial value is entered and the desired field is directly
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:31]:
```suggestion
   filter <search/custom-filters>`. If a partial value is entered, and the desired field is directly
```

File: content/applications/essentials/search.rst [hunk 8]
```diff
@@ -1,100 +1,214 @@
...
+   selected (without selecting the :guilabel:`⏵ (right arrow)`), *all* records containing the typed
+   characters for the selected field will be included.
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:33]:
```suggestion
   characters for the selected field are included.
```

File: content/applications/essentials/search.rst [hunk 9]
```diff
@@ -1,100 +1,214 @@
...
+
+.. _search/filters:
+
+Filters
+=======
+
+Filters are used to select records that meet specific criteria. The default selection of records is
+specific to each view, but can be modified by selecting one or several :ref:`preconfigured filters
+<search/preconfigured-filters>` or by adding a :ref:`custom filter <search/custom-filters>`.
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:42]:
```suggestion
<search/preconfigured-filters>`, or by adding a :ref:`custom filter <search/custom-filters>`.
```

File: content/applications/essentials/search.rst [hunk 10]
```diff
@@ -1,100 +1,214 @@
...
 .. _search/preconfigured-filters:
 
 Preconfigured filters
-=====================
+---------------------
 
-You can modify the default selection of records by clicking :guilabel:`Filters` and selecting one or
-several **preconfigured filters**.
+Modify the default selection of records by clicking the :guilabel:`🔽 (down arrow) icon` from the
+search bar and selecting one or several *preconfigured filters* from the :guilabel:`Filters`
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:50]:
```suggestion
search bar, and selecting one (or several) *preconfigured filters* from the :guilabel:`Filters`
```

File: content/applications/essentials/search.rst [hunk 11]
```diff
@@ -1,100 +1,214 @@
...
+   selected (without selecting the :guilabel:`⏵ (right arrow)`), *all* records containing the typed
+   characters for the selected field will be included.
+
+.. _search/filters:
+
+Filters
+=======
+
+Filters are used to select records that meet specific criteria. The default selection of records is
+specific to each view, but can be modified by selecting one or several :ref:`preconfigured filters
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:41]:
```suggestion
specific to each view, but can be modified by selecting one (or several) :ref:`preconfigured filters
```

File: content/applications/essentials/search.rst [hunk 12]
```diff
@@ -1,100 +1,214 @@
...
+search bar and selecting one or several *preconfigured filters* from the :guilabel:`Filters`
+drop-down menu.
 
 .. example::
-   On the Sales Analysis report, only records at the sales order stage are selected by default.
-   However, you could *also* include records at the quotation stage by selecting
-   :guilabel:`Quotations`. Furthermore, you could *only* include records from a specific year, for
-   example *2022*, by selecting :menuselection:`Order Date --> 2022`.
+   On the *Sales Analysis* report (:menuselection:`Sales --> Reporting --> Sales`), only records
+   that are at the *sales order* stage with an *order date* within the last 365 days are selected by
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:55]:
```suggestion
   that are at the *sales order* stage, with an *order date* within the last 365 days, are selected by
```

File: content/applications/essentials/search.rst [hunk 13]
```diff
@@ -1,100 +1,214 @@
...
+   default.
+
+   To *also* include records at the *quotation* stage, select :guilabel:`Quotations` from the
```
Comment (ksc-odoo) [inline content/applications/essentials/search.rst:58]:
```suggestion
   To also include records at the *quotation* stage, select :guilabel:`Quotations` from the
```

**Review by ksc-odoo — APPROVED** [review]
Hey @samueljlieber -- just finished my Final Review. Great work! Only have some super-minor adjustments that require your attention. Other than that, this looks "good to go" 👍

**Comment by samueljlieber** [comment]
Thank you for your helpful review @ksc-odoo, I implemented your suggestions in ce0ff4e :)

**Comment by samueljlieber** [comment]
@robodoo r+

**Pull request closed without merging**